Source/WebCore/dom/Document.cpp

```diff
@@ -3170,6 +3170,14 @@ bool Document::updateLayoutIfDimensionsOutOfDate(Element& element, OptionSet<Dim
         // Check our containing block chain. If anything in the chain needs a layout, then require a full layout.
         for (CheckedPtr currentRenderer = renderer; currentRenderer && !currentRenderer->isRenderView(); currentRenderer = currentRenderer->container()) {
 
+            if (currentRenderer->isSVGRenderer()) {
+                if (currentRenderer->selfNeedsLayout() || currentRenderer->needsPositionedMovementLayout()) {
```

Comment:
Should we check `dimensionsCheck` the same way we do below with currentBox:
```
    if (dimensionsCheck.containsAny({ DimensionsCheck::Left, DimensionsCheck::Top }) && (currentRenderer->selfNeedsLayout() || currentRenderer->needsPositionedMovementLayout()))
```

Reply:
Yeah, that's a good idea.

I've refactored things a bit so that checks that are common to both types happen first, and then we do box/svg branch for type specific checks.